### PR TITLE
bf: S3C-3783 notification manager crash

### PR DESCRIPTION
**Pull request by naren-rajendran**
Bucket notification specific zookeeper if not created via federation causes notification configuration manager to error out, resulting in a crash. This should ideally exit process but not with a crash. If this happens the process should exit out (without a crash).

### CODE REVIEW

**Comment by naren-rajendran** [comment]
@bert-e reset

File: extensions/notification/NotificationConfigManager.js [hunk 1]
```diff
@@ -418,7 +418,10 @@ class NotificationConfigManager {
      * @return {undefined}
      */
     init(cb) {
-        return this._checkConfigurationParentNode(() => {
+        return this._checkConfigurationParentNode(err => {
+            if (err) {
+                return cb(err);
+            }
```
Comment (miniscruff) [inline extensions/notification/NotificationConfigManager.js:425]: can this be tested? Same with the other.

Comment (naren-rajendran) [reply extensions/notification/NotificationConfigManager.js:425]: I was thinking about it, the error was uncaught in the original case (the bug) and that was the reason for the crash.

Comment (miniscruff) [reply extensions/notification/NotificationConfigManager.js:425]: It should be possible to add a test that causes an error and assert the error is called in the callback. That is usually how I test error checks like this.

Comment (naren-rajendran) [reply extensions/notification/NotificationConfigManager.js:425]: .

**Review by miniscruff — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]
The PR will need to be retargeted to development/7.10.

File: tests/unit/notification/NotificationConfigManager.js [hunk 2]
```diff
@@ -118,6 +124,17 @@ describe('NotificationConfigManager', () => {
...
             ], done);
         });
+
+        it('should return error if checkConfigParentNode fails', done => {
+            sinon.stub(NotificationConfigManager.prototype,
+                '_checkConfigurationParentNode')
+                .callsFake(checkConfigParentNodeStub);
+            const manager = new NotificationConfigManager(params);
+            manager.init(err => {
+                assert(err);
```
Comment (miniscruff) [inline tests/unit/notification/NotificationConfigManager.js:134]:
```suggestion
                assert.ifError(err);
```
https://nodejs.org/api/assert.html#assert_assert_iferror_value

Comment (naren-rajendran) [reply tests/unit/notification/NotificationConfigManager.js:134]: The test is expecting an error to be returned.

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e force_reset

**Comment by naren-rajendran** [comment]
@bert-e approve

**Pull request merged**